### ISSUE TITLE
Looking for a way to create a list of all ToDo's without a due date

### ISSUE THREAD

**Issue by Krit1960** [leibler/dokuwiki-plugin-todo#114]
I want to build a GTD style Todo system, I can create filters to have list with All today, All planned etc... but can't seem to find a way to filter all items with no due date assigned.

**Comment by runout-at** [comment]
no. if any kind of invalid date is given the actual date will be used.
you are welcome to provide a pull request for the feature.
look in syntax/list.php for `function analyseDate($date)`

i have moved my projects away from github to gitlab [0] for some reasons and doesn't contribute here anymore.

[0] https://gitlab.com/dokuwiki-plugins

**Comment by einhirn** [comment]
The work for selecting todo items without due dates assigned will probably have to go into "isRequestedTodo" because a todo is only listed if this returns true. But from a quick look I don't see how exactly I would go and implement it - I guess you'd need to add some kind of parameter to the todolist tag that gets parsed in the "handle"-Function and then react to it in the aforementioned function to check wether the current todo has a due date or not...

**Comment by MacBoeck** [comment]
I was looking for an inverse filter: todos that don't have either start and/or due. So I could have a catch-the-rest list. Feature request or already possible but I missed it?

**Comment by einhirn** [comment]
I'm looking at https://github.com/leibler/dokuwiki-plugin-todo/commit/64495bd0c41df6d83f3ed6170c366656bc104714 which is a blast from the past (committed 2014) - there's something happening with * and ! - it's not in the documentation, but it's in the code... Commit-Message says:
> additional flag for date filters of TODOLIST
attributes affected: startbefore, startafter, duebefore, dueafter
eg:
startafter:! list all tasks where startdate in todo is set
startafter:2014-10-30* list all tasks starting after given date or where no startdate in todo is set

From what I understand something like "dueat:*" should work to list items without a due date... It's a bit unintuitive because `!` usually is connected to negations, while `*` is used for anything, but changing that will introduce a breaking change for those who already use this not yet documented feature. Should consider it anyway.
